test/classes/initializers/errors/errHandling/initWithTryCatch.chpl

```diff
@@ -0,0 +1,24 @@
+// Verifies we currently don't allow initializers to have try statements in
+// their body.
+// We hope to eventually allow it.
+class Foo {
+  var x: int;
+
+  proc init() {
+    x = 10;
+    try {
+      outerFunc();
+    } catch {
+      writeln("Look ma, I caught an error!");
+    }
```

Comment:
I think this case ends up being the most worrisome without banning all error handling structures, as its behavior seems to indicate it would permit field assignment and super.init() calls within the try catch, potentially violating single-initialization of fields and single-super.init() calls (for instance, an early thrown error could prevent the assignment to the field that our pass detected within the later body of the try, and would not insert the appropriate omitted initialization before returning)

Reply:
I don't think you've said yet what this actually does without the "no error handling structures" rule?

Reply:
Without the changes on this branch, it gives the output:
```
Look ma, I caught an error!
{x = 10}
```

I suspect its behavior when I only ban the initializer from being declared as throws should be equivalent, since the catchall does not rethrow anything, so the remainder of the initializer would complete as normal

Reply:
Going back to your earlier point, we could get around this by not allowing try/try! with catch in Phase 1, but allow try! only.  To me, that would imply that we should make the error message more fine-grained as the current one doesn't indicate it would allow some but not others (which is definitely acceptable to me)

Reply:
Nothing about this test makes me think it must be an error... but anyway I feel my viewpoint has been heard & I hope you do something we've recently discussed.

Reply:
Would changing the test to:
```
try {
  outerFunc();
  x = 10;
} catch { ... }
```
clarify my concerns?  I can do that as part of the change

Reply:
Yes that is a much better test showing what should be an error.